test/matchers/proto.go

```diff
@@ -1,13 +1,42 @@
 package matchers
 
 import (
+	"fmt"
 	"reflect"
+	"strings"
 
+	"github.com/go-test/deep"
+	"github.com/golang/mock/gomock"
 	"github.com/golang/protobuf/proto"
 	"github.com/onsi/gomega/format"
 	"github.com/onsi/gomega/types"
 )
 
+// Use this in a gomock EXPECT call e.g.
+// `client.EXPECT().Update(ctx, GomockMatchProto(expected)).Return(nil)`
+func GomockMatchProto(actual interface{}) gomock.Matcher {
```

Comment:
Should we rename this to `GomockMatchPublicFields` since it doesn't only match protos anymore?